addons/stock/models/res_config_settings.py

```diff
@@ -59,6 +59,7 @@ def _onchange_group_stock_multi_locations(self):
     def _onchange_group_stock_production_lot(self):
         if not self.group_stock_production_lot:
             self.group_lot_on_delivery_slip = False
+            self.module_product_expiry = False
```

Comment:
I don't know if it's possible to add a warning or something like that. Because it means we will erase all the expiration data once the settings is untick. If it was a mistake it will require a rollback

Reply:
Okay the warning is already there